test/test_utils.py

```diff
@@ -656,6 +657,7 @@ def test_download_url_to_file(self):
                          SUM_OF_HUB_EXAMPLE)
 
     @retry(URLError, tries=3)
```

Comment:
There's a retry URLError on most of the tests in this file. I just added another layer of retrying to the current test case, but there could be a better subtractive solution

Reply:
maybe we could collect all the possible urllib exceptions into a common_utils decorator `retry_http` or something so we can concentrate it all in one place